Assets/Scripts/MovePlayer.cs

```diff
@@ -0,0 +1,213 @@
+    /// </summary>
+    private enum directions { Down, Up, Left, Right};
+
+    /// <summary>
+    /// The animator attached to the player
+    /// </summary>
+    private Animator playerAnimator;
+    /// <summary>
+    /// Is the player allowed to move? Will be false if paused, in a cutscene, etc.
+    /// </summary>
+    private bool canMove = true;
+    /// <summary>
+    /// Has the player received input to move?
+    /// </summary>
+    private bool shouldMove;
+    /// <summary>
+    /// Is the player in slow terrain?
+    /// </summary>
+    private bool slowMove;
+    /// <summary>
+    /// Is the movement coroutine currently running?
+    /// </summary>
+    private bool moving;
+    /// <summary>
+    /// Which direction is the player trying to move?
+    /// </summary>
+    private directions currentDirection;
+    /// <summary>
+    /// An array for quick reference of directions, in conjunction with the enum
+    /// This prevents the need to write four separate coroutines
+    /// </summary>
+    private Vector3[] movementVectors;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        playerAnimator = GetComponent<Animator>();
+        movementVectors = new Vector3[4] { Vector2.down, Vector2.up, Vector2.left, Vector2.right };
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        GetInput();
+    }
+
+    private void FixedUpdate()
+    {
+        if (canMove && shouldMove && !moving)
+            StartCoroutine(MovementRoutine());
+    }
+
+    private void OnTriggerEnter2D(Collider2D collision)
+    {
+        if (collision.gameObject.tag == slowTerrainTag)
+            slowMove = !slowMove;
+    }
+
+    /// <summary>
+    /// Gets player input
+    /// </summary>
+    private void GetInput()
+    {
+        if (Input.GetAxis("Vertical") < 0)
+        {
+            shouldMove = true;
+            currentDirection = directions.Down;
+        }
+        else if (Input.GetAxis("Vertical") > 0)
+        {
+            shouldMove = true;
+            currentDirection = directions.Up;
+        }
+        else if (Input.GetAxis("Horizontal") < 0)
+        {
+            shouldMove = true;
+            currentDirection = directions.Left;
+        }
+        else if (Input.GetAxis("Horizontal") > 0)
+        {
+            shouldMove = true;
+            currentDirection = directions.Right;
+        }
+        else
+            shouldMove = false;
+    }
+
+    /// <summary>
+    /// Adjusts the player to fit exactly into a tile
+    /// </summary>
+    private void SnapToGrid()
+    {
+        int positionX = Mathf.RoundToInt(transform.position.x);
+        int positionY = Mathf.RoundToInt(transform.position.y);
+        transform.position = new Vector3(positionX, positionY);
+    }
+
+
+    private void PlayWalkAnim(directions movementDirection, bool slowTerrain)
+    {
+        playerAnimator.SetBool("Slow", slowTerrain);
+        switch (movementDirection)
```

Comment:
You can figure out movement direction by looking at the vector 2 for movement input.